pymeasure/instruments/hp/hp3478A.py

```diff
@@ -463,3 +463,109 @@ def trigger(self):
     def trigger(self, value):
         trig_set = self.TRIGGERS[strict_discrete_set(value, self.TRIGGERS)]
         self.write(trig_set)
+
+    @property
+    def calibration_data(self):
+        """Read or write the calibration data as an array of 256 values between 0 and 15.
+
+        The calibration data of an HP 3478A is stored in a 256x4 SRAM that is
+        permanently powered by a 3v Lithium battery. When the battery runs
+        out, the calibration data is lost, and recalibration is required.
+
+        When read, this property fetches and returns the calibration data so that it can be
+        backed up.
+
+        When assigned a value, it similarly expects an array of 256 values between 0 and 15,
+        and writes the values back to the instrument.
+
+        When writing, exceptions are raised for the following conditions:
+
+        * The CAL ENABLE switch at the front of the instrument is not set to ON.
+        * The array with values does not contain exactly 256 elements.
+        * The array with values does not pass a verification check.
+
+        IMPORTANT: changing the calibration data results in permanent loss of
+        the previous data. Use with care!
+
+        """
+        cal_data = []
+        for addr in range(0, 256):
+            # To fetch one nibble: 'W<address>', where address is a raw 8-bit number.
+            cmd = bytes([ord('W'), addr])
+            self.write_bytes(cmd)
+            rvalue = self.read_bytes(1)[0]
+            # 'W' command reads a nibble from the SRAM, but then adds a value of 64 to return
+            # it as an ASCII value.
+            if rvalue < 64 or rvalue >= 80:
+                raise Exception("calibration nibble out of range")
+            cal_data.append(rvalue-64)
+
+        return cal_data
+
+    @calibration_data.setter
+    def calibration_data(self, cal_data):
+        """Setter to write the calibration data.
+
+        """
+
+        if not self.calibration_enabled:
+            raise Exception("CAL ENABLE switch not set to ON")
+
+        self.write_calibration_data(cal_data, True)
+
+    def write_calibration_data(self, cal_data, verify_calibration_data=True):
+        """Method to write calibration data.
+
+        The cal_data parameter format is the same as the 'calibration_data' property.
+
+        Verification of the cal_data array can be bypassed by setting
+        'verify_calibration_data' to False.
+
+        """
+        if verify_calibration_data:
+            if not self.verify_calibration_data(cal_data):
+                raise Exception("cal_data verification fail.")
```

Comment:
You can (but don't have to) combine them:
```suggestion
        if verify_calibration_data and not self.verify_calibration_data(cal_data):
            raise Exception("cal_data verification fail.")
```
Only if the first part (`verify_calibration_data`) is True, the second call is done.

Reply:
Fixed.